### ISSUE TITLE
block_in_if_condition_expr false positive

### ISSUE THREAD

**Issue by iliekturtles** [Manishearth/rust-clippy#516]
The lint gives a false positive for unsafe blocks where the braces are required.

```rust
let a: i32 = 1;

// warning: omit braces around single expression condition
if unsafe { 1u32 == std::mem::transmute(a) } {
    println!("1u32 == a");
}
````
